frame/staking/src/lib.rs

```diff
@@ -1270,6 +1453,100 @@ impl<T: Trait> Module<T> {
 
 	// MUTABLES (DANGEROUS)
 
+	fn do_payout_nominator(who: T::AccountId, era: EraIndex, validators: Vec<(T::AccountId, u32)>)
+		-> DispatchResult
+	{
+		let mut nominator_ledger = <Ledger<T>>::get(&who).ok_or_else(|| Error::<T>::NotController)?;
+		if nominator_ledger.next_reward > era {
+			return Err(Error::<T>::InvalidEraToReward.into());
+		}
+
+		nominator_ledger.next_reward = era + 1;
```

Comment:
era is a user input. I believe you allow for an overflow error here :)

Reply:
One way to solve this issue (beyond additional verification of the inputs), is to track the "last_reward" which is the last reward a user has claimed, rather than the `next_reward`

Reply:
yes the user can reset its next_reward to zero using this error :-/
last_reward is OK but then it must be an Option which is a bit boring in the code,
let's just saturating add. the era index shouldn't wrap around.

Reply:
but if we do saturating then last era index is either non rewardable or you can claim the reward indefinitely, I'll change for option